### ISSUE TITLE
_bar_Unarchiver new hidden property issue

### ISSUE THREAD

**Issue by Fawkes-dev** [Lehona/LeGo#28]
I've migrated from LeGo 2.7.1 to 2.8.0 and my bars disappeared in game save I loaded.
I wonder if it really makes sense to update it from view `b.hidden = (v.isClosed || v.continueClose);` - because when saving game all bars are hidden so this update will always hide them - unless you explicitly call Bar_Show afte game loading.
Would it make sense to assume instead that bar is visible when this property does not exist ? 🤔 
Or maybe I am not using bars properly - let me know.

**Comment by szapp** [comment]
> I wonder if it really makes sense to update it from view `b.hidden = (v.isClosed || v.continueClose);`

Yes. These are properties that will reflect if a view is been closed manually.

> because when saving game all bars are hidden

No. Before LeGo 2.7.0, LeGo bars would only be hidden if the scripts explicitly hid them. When updating from before 2.7.0 to 2.8.0, this should work exactly as intended.

However, as for 2.7.0 and 2.7.1, the handling of *manual hiding* of bars was not correctly supported in these versions. There was a fix for this soon after 2.7.1, but only now made it into the stable version of LeGo. If you had released a mod project in between 2.7.0 and 2.8.0 that relied on manually hidden bars, they should not have worked properly.

In the benefit of the doubt, do you have a minimal code example that worked with 2.7.0 (or 2.7.1) but when loading does not work with 2.8.0?

**Comment by Fawkes-dev** [comment]
Thanks for your reply - unfortunatelly I started using LeGo bars only with 2.7.1 not sure if my code worked in 2.7.0.
This is how I am using LeGo bars:
```c++
if (!Hlp_IsValidHandle(hStaminaBar)) {
	hStaminaBar = Bar_Create (GothicBar@);

	//180, 20
	Bar_ResizePxl (hStaminaBar, 180, 10);

	Bar_SetBarTexture (hStaminaBar, "Bar_Misc.tga");
};
```
and then updating it:
```c++
Bar_SetMax (hStaminaBar, max);
Bar_SetValue (hStaminaBar, value);
```

**Comment by szapp** [comment]
If you had released a mod with LeGo 2.7.1, you can correct the then-hidden bars loaded from savegames by calling `Bar_Show`. You can do so anywhere in your code that is executed after loading. That will set the bar from manually hidden back to being auto-hidden and auto-shown. After that the bar will behave as expected again. In your particular code example, this here may be an option:
```D
if (!Hlp_IsValidHandle(hStaminaBar)) {
    hStaminaBar = Bar_Create (GothicBar@);

    //180, 20
    Bar_ResizePxl (hStaminaBar, 180, 10);

    Bar_SetBarTexture (hStaminaBar, "Bar_Misc.tga");
} else {
    Bar_Show(hStaminaBar);
};
```

**Comment by Fawkes-dev** [comment]
Thank you :)
